docs/developer/extending/apps/building-payment-app.mdx

```diff
@@ -0,0 +1,856 @@
+
+    return res.status(200).json({
+      result: "CHARGE_SUCCESS",
+      amount: payload.action.amount,
+      pspReference: crypto.randomUUID(),
+    });
+  }
+);
+
+export const config = {
+  api: {
+    bodyParser: false,
+  },
+};
+// highlight-end
+```
+
+Once again, our handler is simple. We only log the payload and return the `CHARGE_SUCCESS` event status, no questions asked.
+
+:::info
+
+A real-world `TRANSACTION_PROCESS_SESSION` webhook handler might:
+
+- Call the payment provider API to confirm the payment status.
+- Return either `CHARGE_SUCCESS` or `CHARGE_FAILURE` based on the response.
+
+:::
+
+#### Error Handling
+
+If the handler performs a logic that could fail, we should try to catch the error and, in that case, return the `CHARGE_FAILURE` event status:
+
+```ts
+export default transactionProcessSessionWebhook.createHandler(
+  (req, res, ctx) => {
+    const { payload, event, baseUrl, authData } = ctx;
+
+    console.log("Transaction Process Session payload:", payload);
+
+    try {
+      doSomethingThatCanFail(); // This function can throw an error
+
+      return res.status(200).json({
+        result: "CHARGE_SUCCESS",
+        amount: payload.action.amount,
+        // TODO: CHECK IF THESE ARE THE SAME PSP REFERENCES
+        pspReference: crypto.randomUUID(),
+      });
+    } catch (error) {
+      return res.status(200).json({
+        result: "CHARGE_FAILURE",
+        amount: payload.action.amount,
+        pspReference: crypto.randomUUID(),
+      });
+    }
+  }
+);
+```
+
+#### Updating the App Manifest
+
+The last step is to update the App Manifest with the new webhook:
+
+```ts
+// src/pages/api/manifest.ts
+import { createManifestHandler } from "@saleor/app-sdk/handlers/next";
+import { AppManifest } from "@saleor/app-sdk/types";
+import { transactionInitializeSessionWebhook } from "./webhooks/transaction-initialize-session";
+// highlight-next-line
+import { transactionProcessSessionWebhook } from "./webhooks/transaction-process-session";
+
+export default createManifestHandler({
+  async manifestFactory({ appBaseUrl }) {
+    // ...
+    const manifest: AppManifest = {
+      name: "Dummy Payment App",
+      webhooks: [
+        transactionInitializeSessionWebhook.getWebhookManifest(appBaseUrl),
+        // highlight-next-line
+        transactionProcessSessionWebhook.getWebhookManifest(appBaseUrl),
+      ],
+      // ...
+    };
+  },
+});
+```
+
+Once again, **remember to update the app's webhooks or reinstall the app in Saleor**. Otherwise, the new webhook handler won't be called on `transactionProcess`.
+
+:::warning
+
+When installing the app back, you may get "**App with the same identifier is already installed**" error.
+
+Saleor throws this error because the process of fully uninstalling the app is handled by an asynchronous Saleor worker. This means it can take some time before you can install the app with the same identifier again.
```

Comment:
Also worth mentioning, that in the case of using Saleor locally, you should have active background workers (celery) and celery-beat